src/components/OauthRedirectHandler/index.tsx

```diff
@@ -0,0 +1,61 @@
+import type { ProviderType } from '@/types/auth';
+import axios from 'axios';
+import { useEffect } from 'react';
+import { useNavigate, useParams, useSearchParams } from 'react-router-dom';
+
+const OauthRedirectHandler = () => {
+  const { provider } = useParams<{ provider: ProviderType }>();
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+  const code = searchParams.get('code');
+  const state = searchParams.get('state');
+
+  useEffect(() => {
+    const login = async () => {
+      if (!code) {
+        navigate('/login');
+        return;
+      }
+
+      try {
+        const requestData =
+          provider === 'naver'
+            ? { provider, authorizationCode: code, state }
+            : { provider, authorizationCode: code };
+
+        const response = await axios.post(
+          `${import.meta.env.VITE_API_BASE_URL}/auth/login`,
+          requestData,
+          {
+            headers: {
+              'Content-Type': 'application/json',
+            },
+          }
+        );
+
+        const { needSignup, accessToken } = response.data;
+
+        if (needSignup) {
+          navigate('/signup');
+        } else {
+          // TODO: accessToken 저장 로직 수정
+          localStorage.setItem('accessToken', accessToken);
+          navigate('/home/subscription');
+        }
+      } catch (error) {
+        console.error('로그인에 실패했습니다.', error);
+        navigate('/login');
+      }
+    };
+
+    login();
+  });
+
+  return (
+    <div className="flex flex-col items-center justify-center h-screen">
+      로그인 처리 중입니다...
+    </div>
+  );
+};
+
+export default OauthRedirectHandler;
```

Comment:
저희 export를 
```suggestion
export const OauthRedirectHandler = () ...
```
이렇게 하기로 얘기한거 아닐까요?
default를 하려면 이렇게 밑에서 따로 해주는게 맞는데, 그럼 컨벤션을 모두 밑에서 export 하는걸로 하는게 맞을 것 같아요!

Reply:
> default를 하려면 이렇게 밑에서 따로 해주는게 맞는데, 그럼 컨벤션을 모두 밑에서 export 하는걸로 하는게 맞을 것 같아요!

`export`를 하게되면 `import`할 때 `import OauthRedirectHandler from './components/OauthRedirectHandler/OauthRedirectHandler.tsx';` 이런식으로 중복되는 경우가 생겨 `export default`를 사용했습니다.

어떻게 통일하는게 좋을까요? @leesm0218 @smileboy0014

Reply:
만약 default를 쓸거면, 저는 그냥 밑에서 다 export 해주는걸로 통일하는게 맞는 것 같습니다!

Reply:
index.tsx가 배럴 역할을 하게 하면 될 것 같습니다.

Reply:
배럴 역할을 하게 하면 된다는게 어떤 식인지 설명을 드려도 괜찮을까요?

Reply:
지금 현수님이 걱정하시는게, 저걸 index가 아닌 OauthRedirectHandler.tsx 파일로 만들면

import OauthRedirectHandler from './components/OauthRedirectHandler/OauthRedirectHandler.tsx'

이렇게 길게 적어야 된다고 그러시는것 같은데,

OauthRedirectHandler.tsx 로 빼도 index 에서 export 해주면 똑같이 

import OauthRedirectHandler from './components/OauthRedirectHandler/

처럼 뒤에 폴더명까지만 작성해서 불러올 수 있을겁니다.